src/uuid/variant.cr

```diff
@@ -0,0 +1,62 @@
+struct UUID
+  # UUID variants.
+  enum Variant
+    # Unknown (ie. custom, your own).
+    Unknown
+
+    # Reserved by the NCS for backward compatibility.
+    NCS
+
+    # As described in the RFC4122 Specification (default).
+    RFC4122
+
+    # Reserved by Microsoft for backward compatibility.
+    Microsoft
+
+    # Reserved for future expansion.
+    Future
+  end
+
+  # Returns UUID variant based on provided 8th `byte` (0-indexed).
+  def self.byte_variant(byte : UInt8)
```

Comment:
We don't need the `byte_variant` class methods. They should be inlined into `#variant` and `#variant=`.